todo/src/Components/UserList.tsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable jsx-a11y/img-redundant-alt */
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import InfiniteScroll from "react-infinite-scroll-component";
+import { UserListProps } from "../Todo";
+import { makeStyles } from "@material-ui/styles";
+
+const useStyles = makeStyles({
+  container: {
+    length: "30px",
+    width: "600px",
+    alignItems: "center",
+  },
+  cards: {
+    display: "flex",
+    flexDirection: "row",
+    columnGap: "10px",
+    position: "relative",
+    height: "130px",
+  },
+  image: {
+    borderRadius: "90px",
+    width: "70px",
+    alignContent: "center",
+  },
+  name: {
+    alignContent: "center",
+    position: "absolute",
+    top: "0px",
+    left: "90px",
+    display: "flex",
+    flexDirection: "column",
+    rowGap: "1px",
+  },
+  scrollbar: {
+    height: "10px",
+    overflow: " auto",
+  },
+  email: {
+    fontStyle: "none",
+    position: "absolute",
+    top: "53px",
+    fontSize: "17px",
+  },
+  hr: {
+    width: "450px",
+    position: "absolute",
+    top: "45px",
+    color: "grey",
+  },
+});
+const UserList = () => {
+  const classes = useStyles();
+  const PER_PAGE = 2;
+  const apiPath = "https://reqres.in/api/users?";
+  const [Lists, setLists] = useState<UserListProps[]>([]);
+  const [length, setLength] = React.useState<number>(0);
+  const getList = () => {
+    const pageNo = Math.ceil(Lists.length / PER_PAGE) + 1;
+    const queryParam = "page=" + pageNo + "&per_page=" + PER_PAGE;
+    const finalUrl = apiPath + queryParam;
+    axios
+      .get(finalUrl)
+      .then((res) => {
+        const apiRes = res.data.data;
+        setLength(res.data.total);
+        const mergeData = [...Lists, ...apiRes];
+        setLists(mergeData);
+      })
+      .catch((err) => {
+        console.error("Error while Loading", err);
+      });
+  };
+  
+  useEffect(() => {
+    setTimeout(() => getList(), 1000);
+  }, []);
+
+  const fetchMoreData = () => {
+    setTimeout(() => getList(), 1000);
```

Comment:
Why setTimeout ?

Reply:
I used to observe the execution of user list and "Loading..."  one after the other.

Reply:
using setTimeout means that you have not understood how the async process works.  In this case please remove the setTimeout and implement

Reply:
Ok sir